emu.c

```diff
@@ -294,10 +288,18 @@ int main(int parc, char ** pars)
     emu.mUpperData   = malloc(128);
     emu.mSFR         = malloc(128);
     emu.except       = &emu_exception;
-    emu.sfrread      = &emu_sfrread;
-    emu.sfrwrite     = &emu_sfrwrite;
+    emu.sfrread      = malloc(128 * sizeof(em8051sfrread));
+    emu.sfrwrite     = malloc(128 * sizeof(em8051sfrwrite));
```

Comment:
Yeah, not feeling too hot about these mallocs. Just make a 128x array.

Reply:
Done in 508ba99aa4f5de2dc9be8fcdc968357582116e2a for all fixed-sized areas.

I also went further in 1154a0add856c29662274dea757e22b866604f7a and used the variable size to allocate memory.
And I used `calloc()` as my testing showed that I forgot to zero-fill the `emu.sfrread` callback table in my initial commit.